### PR TITLE
[DOCS] Security updates.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by leonardoalt — DISMISSED** [review]

File: docs/security-considerations.rst [hunk 1]
```diff
@@ -246,12 +246,8 @@ implications, there might be another issue buried beneath it.
 Any compiler warning we issue can be silenced by slight changes to the
 code.
 
-Also try to enable the "0.5.0" safety features as early as possible
-by adding ``pragma experimental "v0.5.0";``. Note that in this case,
-the word ``experimental`` does not mean that the safety features are in any
-way risky, it is just a way to enable some features that are
-not yet part of the latest version of Solidity due to backwards
-compatibility.
+Always use the latest version of the compiler, to be notified about all recently
```
Comment (redsquirrel) [inline docs/security-considerations.rst]: That comma is unnecessary.

Comment (chriseth) [reply docs/security-considerations.rst]: thanks!

File: docs/security-considerations.rst [hunk 2]
```diff
@@ -136,15 +136,15 @@ Sending and Receiving Ether
 - If a contract receives Ether (without a function being called), the fallback function is executed.
   If it does not have a fallback function, the Ether will be rejected (by throwing an exception).
   During the execution of the fallback function, the contract can only rely
-  on the "gas stipend" it is passed (2300 gas) being available to it at that time. This stipend is not enough to access storage in any way.
+  on the "gas stipend" it is passed (2300 gas) being available to it at that time. This stipend is not enough to modify storage.
```
Comment (axic) [inline docs/security-considerations.rst]: I'd add a comment: do not take this for granted though, the stipend is subject to change with future hard forks.

File: docs/security-considerations.rst [hunk 3]
```diff
@@ -305,6 +301,13 @@ of "failsafe" mode, which, for example, disables most of the features, hands ove
 control to a fixed and trusted third party or just converts the contract into
 a simple "give me back my money" contract.
 
+Ask for Peer Review
+===================
+
+The more eyes are used on a piece of code, the more issues are found.
+This is especially beneficial for small numbers of eyes.
```
Comment (axic) [inline docs/security-considerations.rst]: Not sure about these two sentences about eyes 👀

Comment (erak) [reply docs/security-considerations.rst]: I'd also talk about people doing reviews. Or maybe use "another pair of eyes"?

File: docs/security-considerations.rst [hunk 4]
```diff
@@ -246,12 +246,8 @@ implications, there might be another issue buried beneath it.
 Any compiler warning we issue can be silenced by slight changes to the
 code.
 
-Also try to enable the "0.5.0" safety features as early as possible
-by adding ``pragma experimental "v0.5.0";``. Note that in this case,
-the word ``experimental`` does not mean that the safety features are in any
-way risky, it is just a way to enable some features that are
-not yet part of the latest version of Solidity due to backwards
-compatibility.
+Always use the latest version of the compiler to be notified about all recently
```
Comment (erak) [inline docs/security-considerations.rst:270]: The grammar seems to be a bit off here. Perhaps change it to ``Always use the latest version of the compiler to be notified about all warnings introduced recently``.

**Comment by erak** [comment]
I'd suggest to add a section about arithmetic over / underflows to the pitfalls. We could start off with something like: 

"An over/under flow occurs when an operation is performed that requires a fixed size variable to store a number (or piece of data) that is outside the range of the variable's data type. For example, subtracting ``1`` from a ``uint8`` (unsigned integer of ``8`` bits, i.e. only positive) variable that stores ``0`` as it's value, will result in the number ``255``. This is an underflow. "

I think a real-world example could be beneficial and we could also recommend the usage of ``SafeMath.sol``.

**Comment by chriseth** [comment]
@bit-shift created https://github.com/ethereum/solidity/pull/5111 about overflows

File: docs/security-considerations.rst (hunk 2 above)
Comment (chriseth) [reply docs/security-considerations.rst]: Actually I would strongly object any rule change that allows storage modifications after a `.transfer`.

Comment (chriseth) [reply docs/security-considerations.rst]: But yeah, we can still warn users about it :)

File: docs/security-considerations.rst [hunk 5]
```diff
@@ -305,6 +302,12 @@ of "failsafe" mode, which, for example, disables most of the features, hands ove
 control to a fixed and trusted third party or just converts the contract into
 a simple "give me back my money" contract.
 
+Ask for Peer Review
+===================
+
+The people examine a piece of code, the more issues are found.
```
Comment (axic) [inline docs/security-considerations.rst]: The more people?

Comment (chriseth) [reply docs/security-considerations.rst]: Argh! I hope the documentation marathon will end soon...

Comment (chriseth) [reply docs/security-considerations.rst]: Fixed.

File: docs/security-considerations.rst (hunk 2 above)
Comment (axic) [reply docs/security-considerations.rst]: There is a proposal to increase it already (albeit only with making all kinds of LOGs working even in proxies): https://eips.ethereum.org/EIPS/eip-1285

**Review by axic — DISMISSED** [review]

**Pull request merged**